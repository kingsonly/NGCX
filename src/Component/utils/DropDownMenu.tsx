import React, {useState, FC} from "react";


interface DropDownMenuProps {
    //boolean to always open ddm (for presentation)
    forceOpen?: boolean;
    label?: string;
    withDivider?: boolean;
    icon?: JSX.Element;
    items: DDMItem[];
    variant: string;
    // styles: String
}

export interface DDMItem {
    icon?: JSX.Element;
    label: string;
    desc?: string;
    link?: string;
}


const styles = {
    'withBg': 'bg-ncgx-green text-white font-bold rounded-md',
    'noBg': 'bg-white text-gray-600 font-bold',
    'searchDdm': 'font-light bg-white text-gray-600 font-medium flex-1 w-40'
}

const DropDownMenu:FC<DropDownMenuProps> = (props) => {
    const [isOpen, setIsOpen] = useState(false);
    return (
        <div className="relative inline-block text-left">
            <div>
                <button
                    type="button"
                    onClick={() => setIsOpen(!isOpen)}
                    // @ts-ignore
                    className={` ${styles[props.variant]} flex items-center justify-center rounded-md px-4 py-2 text-sm hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-offset-gray-100 focus:ring-gray-500`}
                    id="options-menu"
                >

                    {props.label}

                    <span className="ml-3">{props.icon}</span>
                    
                </button>
            </div>

            {(props.forceOpen || isOpen) && (
                <div className="origin-top-right absolute right-0 mt-2 w-56 rounded-md shadow-lg bg-red-900 dark:bg-gray-800 ring-1 ring-black ring-opacity-5">
                    <div
                        className={`py-1 ${props.withDivider ? 'divide-y divide-gray-100' : ''}`}
                        role="menu"
                        aria-orientation="vertical"
                        aria-labelledby="options-menu"
                    >
                        {props.items.map((item) => {
                            return (
                                <a
                                    key={item.label}
                                    href={item.link || '#'}
                                    className={`${
                                        item.icon ? 'flex items-center' : 'block'
                                    } block px-4 py-2 text-md text-gray-700 hover:bg-gray-100 hover:text-gray-900 dark:text-gray-100 dark:hover:text-white dark:hover:bg-gray-600`}
                                    role="menuitem"
                                >
                                    {item.icon}

                                    <span className="flex flex-col">
                                        <span>{item.label}</span>
                                        {item.desc && <span className="text-gray-400 text-xs">{item.desc}</span>}
                                    </span>
                                </a>
                            );
                        })}
                    </div>
                </div>
            )}
        </div>
    );
};





export default DropDownMenu