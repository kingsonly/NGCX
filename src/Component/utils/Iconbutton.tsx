import React, {FC} from 'react'
import { FiX, FiMenu } from "react-icons/fi"

interface IconbuttonProps {
    isOpen : Boolean,
    onToggle : Function
}

const Iconbutton:FC<IconbuttonProps> = ({isOpen}) => {
  return (
    <button
  >
      {isOpen ? <FiX size={16} /> : <FiMenu size={16} />}
  </button>
  )
}

export default Iconbutton