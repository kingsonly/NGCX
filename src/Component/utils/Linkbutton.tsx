import React, {FC} from 'react'

interface LinkbuttonProps {
    text: string
}

const Linkbutton: FC<LinkbuttonProps> = ({text}) => {
  return (
    <div>
        <button className='text-sm font-semibold'><a href='#'>{text}</a></button>
    </div>
  )
}

export default Linkbutton