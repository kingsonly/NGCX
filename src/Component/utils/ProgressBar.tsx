import React from 'react'

const ProgressBar = () => {
  return (  
    <div>
        <div className="bg-white rounded-lg w-72 block m-auto">
            <div className="w-full h-2 bg-gray-200 rounded-full mt-3">
                <div className="w-3/4 h-full text-center text-xs text-white bg-green-500 rounded-full">
                </div>
            </div>
        </div>
    </div>
  )
}

export default ProgressBar