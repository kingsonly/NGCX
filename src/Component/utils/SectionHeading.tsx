import React,{FC} from 'react'

interface SectionHeadingProps {
    text: string
}

const SectionHeading:FC<SectionHeadingProps> = ({text}) => {
  return (
    <div className='text-ncgx-text-gray font-bold text-lg my-10' >{text}</div>
  )
}

export default SectionHeading