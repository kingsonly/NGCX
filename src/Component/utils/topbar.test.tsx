/**
 * @jest-environment jsdom
 */

 import '@testing-library/jest-dom'

import renderer from "react-test-renderer";
import {
  render,
  screen,
  fireEvent,
  queryByLabelText,
  queryByTestId,
} from "@testing-library/react";

import React from "react";
import DropDownMenu from './DropDownMenu';
import Iconbutton from './Iconbutton';
import { FiMenu, FiX } from 'react-icons/fi';
import Linkbutton from './Linkbutton';
import ProgressBar from './ProgressBar';
import SectionHeading from './SectionHeading';


describe("Test Utils for top bar component", () => {
  test("renders Iconbutton correctly", () => {
    const component = render(<Iconbutton  onToggle={Function}  isOpen={true} ></Iconbutton>)
    expect(component).toMatchSnapshot();
  });

  test("renders LinkButton correctly", () => {
    const component = render(<Linkbutton text='Test'></Linkbutton>)
    expect(component).toMatchSnapshot();
  });

  test("renders Progress correctly", () => {
    const component = render(<ProgressBar  ></ProgressBar>)
    expect(component).toMatchSnapshot();
  });

  test("renders Best Seller correctly", () => {
    const component = render(<SectionHeading text='Test' ></SectionHeading>)
    expect(component).toMatchSnapshot();
  });

});
