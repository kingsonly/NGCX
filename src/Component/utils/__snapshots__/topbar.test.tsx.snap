// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Utils for top bar component renders Best Seller correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="text-ncgx-text-gray font-bold text-lg my-10"
      >
        Test
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="text-ncgx-text-gray font-bold text-lg my-10"
    >
      Test
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test Utils for top bar component renders Iconbutton correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button>
        <svg
          fill="none"
          height="16"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="18"
            x2="6"
            y1="6"
            y2="18"
          />
          <line
            x1="6"
            x2="18"
            y1="6"
            y2="18"
          />
        </svg>
      </button>
    </div>
  </body>,
  "container": <div>
    <button>
      <svg
        fill="none"
        height="16"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="18"
          x2="6"
          y1="6"
          y2="18"
        />
        <line
          x1="6"
          x2="18"
          y1="6"
          y2="18"
        />
      </svg>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test Utils for top bar component renders LinkButton correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <button
          class="text-sm font-semibold"
        >
          <a
            href="#"
          >
            Test
          </a>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <button
        class="text-sm font-semibold"
      >
        <a
          href="#"
        >
          Test
        </a>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test Utils for top bar component renders Progress correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="bg-white rounded-lg w-72 block m-auto"
        >
          <div
            class="w-full h-2 bg-gray-200 rounded-full mt-3"
          >
            <div
              class="w-3/4 h-full text-center text-xs text-white bg-green-500 rounded-full"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="bg-white rounded-lg w-72 block m-auto"
      >
        <div
          class="w-full h-2 bg-gray-200 rounded-full mt-3"
        >
          <div
            class="w-3/4 h-full text-center text-xs text-white bg-green-500 rounded-full"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
